### DISCUSSION TITLE
Acessing the images catalog using Kubernetes

### DISCUSSION THREAD

**Discussion by thaimunhoz** [microsoft/PlanetaryComputer#316] [Q&A]
Hello.

I'm currently trying to access and process Sentinel-2 images using the Planetary catalog. In order to improve the scalable computing i'm using Dask in a kubernete. However, even with the use of Dask, accessing the images has taken a significant amount of time. Could anyone tell me how to improve processing performance outside of the Microsoft Hub?

I've tried different configurations in the K8S, but no success.

P.S.: I'm using a dask gateway cluster with the .adapt function (min = 4, max = 300).

This is the part of my code where I access the images:

          `search = catalog.search(collections = ["sentinel-2-l2a"],
                                datetime = start_date + '/' + end_date,
                                bbox = bbox,
                                )
        
        items_dict = search.item_collection_as_dict()['features']
        
        item_collection = pystac.ItemCollection(items_dict)
    
        dc = stac_load(item_collection,
                       measurements=['B04', 'B08'],
                       chunks={"x": 2000, "y": 2000},
                       bbox=bbox,
                       patch_url=pc.sign)

        epsg_ = dc['spatial_ref'].values
        geo_df = polygons_within.to_crs(epsg=epsg_)  # Geodataframe dos pontos
        lon = pd.Series(geo_df.geometry.x)
        lat = pd.Series(geo_df.geometry.y)
        NearGrid_v10 = dc.sel(y=lat.to_xarray(), x=lon.to_xarray(), method='nearest').reset_coords(drop=True)

        ndvi = ((NearGrid_v10['B08'] - NearGrid_v10['B04']) / (NearGrid_v10['B08'] + NearGrid_v10['B04'])).clip(0, 1).rename("ndvi")

        ndvi_masked = ndvi

        time_values = ndvi_masked['time'].values
   
        ndvi_values = ndvi_masked.values.reshape(len(time_values), -1)`

**Comment by TomAugspurger** [comment]
I'll typically use tools like [snakeviz](https://jiffyclub.github.io/snakeviz/) or [line-profile](https://github.com/pyutils/line_profiler) to see where time is being spent.

> Could anyone tell me how to improve processing performance outside of the Microsoft Hub?

The profiling tools will tell you where time is being spent. If most of the time is spent on reading data (as I'm guessing) then the most important thing would be to put your compute in the same Azure region as the data (West Europe in the case of sentinel-2-l2a). That will maximize the bandwidth between the storage nodes and the Blob Storage service.
